segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -531,9 +539,22 @@ private boolean handleClosedLedgers(List<Write> writes) {
             }
 
             // Write likely failed because of LedgerClosedException. Need to check the LastAddConfirmed for each
-            // involved Ledger and see if the write actually made it through or not.
+            // involved Ledger and see if the Write actually made it through or not.
```

Comment:
Is it is possible that it won't have "made it" when this is called but could be later?

Reply:
The `lastAddEntryConfirmed` is invoked after the ledger is closed. My understanding is that, if the ledger is closed, no further write can succeed on that ledger. And the rollover mechanism is at the `BookkeeperLog` level, that creates a new ledger to continue writing to. So the Bookkeeper client cannot inadvertently retry the write to the new ledger, as it does not know about it. Therefore, I would say that it is not possible to get a successful write already initiated after this point.